template.js

```diff
@@ -0,0 +1,199 @@
+const isLoggingEnabled = determinateIsLoggingEnabled();
+const traceId = getRequestHeader('trace-id');
+
+const requestOrigin = getRequestHeader('Origin');
+const requestMethod = getRequestMethod();
+const requestBody = getRequestBody();
+const requestUrl = generateRequestUrl();
+const requestHeaders = generateRequestHeaders();
+
+if (isLoggingEnabled) {
+    logToConsole(JSON.stringify({
+        'Name': 'Exponea',
+        'Type': 'Request',
+        'TraceId': traceId,
+        'RequestOrigin': requestOrigin,
+        'RequestMethod': requestMethod,
+        'RequestUrl': requestUrl,
+        'RequestHeaders': requestHeaders,
+        'RequestBody': requestBody,
+    }));
+}
+
+sendHttpRequest(requestUrl, {method: requestMethod, headers: requestHeaders}, requestBody).then((result) => {
+    if (isLoggingEnabled) {
+        logToConsole(JSON.stringify({
+            'Name': 'Exponea',
+            'Type': 'Response',
+            'TraceId': traceId,
+            'ResponseStatusCode': result.statusCode,
+            'ResponseHeaders': result.headers,
+            'ResponseBody': result.body,
+        }));
+    }
+
+    for (const key in result.headers) {
+        if (key === 'set-cookie') {
+            setResponseCookies(result.headers[key]);
+        } else {
+            setResponseHeader(key, result.headers[key]);
+        }
+    }
+
+    setResponseBody(result.body);
+    setResponseStatus(result.statusCode);
+
+    if (requestOrigin) {
+        setResponseHeader('access-control-allow-origin', requestOrigin);
+        setResponseHeader('access-control-allow-credentials', 'true');
+    }
+
+    returnResponse();
+});
+
+
+function generateRequestUrl() {
+    let url = data.targetAPI + getRequestPath();
+    const queryParams = getRequestQueryString();
+
+    if (queryParams) url = url + '?' + queryParams;
+
+    return url;
+}
+
+function generateRequestHeaders() {
+    let headers = {};
+    let cookies = [];
+
+    for (let i = 0; i < headerWhiteList.length; i++) {
+        let headerName = headerWhiteList[i];
+        let headerValue = getRequestHeader(headerName);
+
+        if (headerValue) {
+            headers[headerName] = getRequestHeader(headerName);
+        }
+    }
+
+    headers.cookie = '';
+
+    for (let i = 0; i < cookieWhiteList.length; i++) {
+        let cookieName = cookieWhiteList[i];
+        let cookieValue = getCookieValues(cookieName);
+
+        if (cookieValue && cookieValue.length) {
+            cookies.push(cookieName + '=' + cookieValue[0]);
+        }
+    }
+
+    headers.cookie = cookies.join('; ');
+    headers['X-Forwarded-For'] = getRemoteAddress();
+
+    return headers;
+}
+
+function setResponseCookies(setCookieHeader) {
+    for (let i = 0; i < setCookieHeader.length; i++) {
+        let setCookieArray = setCookieHeader[i].split('; ').map(pair => pair.split('='));
+        let setCookieJson = '';
+
+        for (let j = 1; j < setCookieArray.length; j++) {
```

Comment:
nit: wouldn't it be easier to create the json object directly (i.e. using `.reduce`) instead of creating a string a then `JSON.parse` it?